src/vis/bar/BarVis.tsx

```diff
@@ -0,0 +1,212 @@
+
+interface BarVisProps {
+    config: IBarConfig;
+    optionsConfig?: {
+        group?: {
+            enable?: boolean;
+            customComponent?: React.ReactNode;
+        },
+        multiples?: {
+            enable?: boolean;
+            customComponent?: React.ReactNode;
+        },
+        direction?: {
+            enable?: boolean;
+            customComponent?: React.ReactNode;
+        },
+        groupingType?: {
+            enable?: boolean;
+            customComponent?: React.ReactNode;
+        },
+        display?: {
+            enable?: boolean;
+            customComponent?: React.ReactNode;
+        }
+    };
+    extensions?: {
+        prePlot?: React.ReactNode;
+        postPlot?: React.ReactNode;
+        preSidebar?: React.ReactNode;
+        postSidebar?: React.ReactNode;
+    };
+    columns: (NumericalColumn | CategoricalColumn) [];
+    setConfig: (config: IVisConfig) => void;
+    scales: Scales;
+}
+
+const defaultConfig = {
+    group: {
+        enable: true,
+        customComponent: null,
+    },
+    multiples: {
+        enable: true,
+        customComponent: null,
+    },
+    direction: {
+        enable: true,
+        customComponent: null,
+    },
+    groupType: {
+        enable: true,
+        customComponent: null,
+    },
+    display: {
+        enable: true,
+        customComponent: null,
+    }
+};
+
+const defaultExtensions = {
+    prePlot: null,
+    postPlot: null,
+    preSidebar: null,
+    postSidebar: null
+};
+
+export function BarVis({
+    config,
+    optionsConfig,
+    extensions,
+    columns,
+    setConfig,
+    scales
+}: BarVisProps) {
+    const mergedOptionsConfig = useMemo(() => {
+        return merge({}, defaultConfig, optionsConfig);
+    }, []);
+
+    const mergedExtensions = useMemo(() => {
+        return merge({}, defaultExtensions, extensions);
+    }, []);
+
+    const traces: PlotlyInfo = useMemo(() => {
+        return createBarTraces(columns, config, scales);
+    }, [columns, config, scales]);
+
+    const layout = useMemo(() => {
+        const layout = {
+            showlegend: true,
+            legend: {
+                itemclick: false,
+                itemdoubleclick: false
+            },
+            autosize: true,
+            grid: {rows: traces.rows, columns: traces.cols, xgap: .3, pattern: 'independent'},
+            shapes: [],
+            violingap: 0,
+            barmode: config.groupType === EBarGroupingType.STACK ? 'stack' : 'group'
+        };
+
+        return beautifyLayout(traces, layout);
+    }, [traces, config.groupType]);
+
+    return (
+        <div className="d-flex flex-row w-100 h-100">
+            <div className="position-relative d-flex justify-content-center align-items-center flex-grow-1">
+                {mergedExtensions.prePlot}
+                {traces.plots.length > 0 ?
+                    (<Plot
+                        divId={'plotlyDiv'}
```

Comment:
If this id is added as `id` attribute to the DOM element then it must be unique, since it is most likely that multiple visualizations used on the same page. Please also change this in the other visualizations.

Reply:
done